src/Microsoft.DotNet.Build.Tasks/Targets/publishtest.targets

```diff
@@ -35,16 +30,25 @@
     </TestTargetFramework>
   </ItemGroup>
 
-  <Target Name="CopyTestToTestDirectory"
-    Inputs="@(PackagesConfig);$(TestRuntimePackageConfig);@(ReferenceCopyLocalPaths);@(Content);@(IntermediateAssembly);@(_DebugSymbolsIntermediatePath)"
-    Outputs="$(TestPath)%(TestTargetFramework.Folder)\*.*"
-    Condition="'$(CopyTestToTestDirectory)'=='true'">
-    
+  <ItemGroup>
+    <PackagesConfigs Include="$(ProjectPackagesConfigFile)" />
+    <PackagesConfigs Include="$(TestRuntimePackageConfig)" />
+  </ItemGroup>
+
+  <Target Name="DiscoverTestInputs" BeforeTargets="RunTestsForProject">
     <ItemGroup>
-      <PackagesConfigs Include="$(ProjectPackagesConfigFile)" />
-      <PackagesConfigs Include="$(TestRuntimePackageConfig)" />
+      <TestInputs Include="@(PackagesConfigs)" />
```

Comment:
Two more comments\questions:
1) Is it worth putting this target with RunTestsForProject or just keeping it here? Just a question I'm not sure which I like better.
2) We should probably rename this to be more specific. Something like RunTestsForProjectTargetInputs to make sure it doesn't class with other things that maybe TestInputs.